### ISSUE TITLE
Fix Broken CircleCI Builds

### ISSUE THREAD

**Issue by DanRStevens** [OutpostUniverse/OP2MapImager#37]
We've had a few failed CircleCI builds for this project in a row now. The code itself seems fine, and compiles cleanly on my computer. The issue seems to be related to the CircleCI environment.

My first thought was incorrect Git submodule reference. There have been some submodule changes recently that looked related to the build errors. I've been unable to reproduce the error on my machine.

Related, I briefly looked into push order problems, where perhaps an update to OP2MapImager might have been pushed before a dependent update to OP2Utility, though that doesn't seem to be the case either. In particular, re-running a failed job fails again, despite it building cleanly on my machine.

My second though was some kind of caching on the build machines, though it doesn't appear we have any sort of caching enabled for CircleCI based on the config file.

----

Failed builds:
[CircleCI OP2MapImager status](https://circleci.com/gh/OutpostUniverse/OP2MapImager)
Currently, from build #47 to build #54 we have mostly failures. Many of these are for the master branch.

**Comment by DanRStevens** [comment]
Ahh, I think the problem is with the Git checkout:
```
    steps:
      - checkout
      - run: git submodule init
      - run: git submodule update --remote
```

The `git submodule update --remote` command is causing it to use a newer version of the branch, rather than the super project's recorded SHA-1 value.

**Comment by DanRStevens** [comment]
Ok, I've been able to reproduce the error. Using the CircleCI logs, I found out which submodule version was used. I ran `git checkout e109237c906a72266d57463fccc877d353f7f29e` within the OP2Utility folder, and then from the main OP2MapImager folder I ran `make`. The build fails as in the CircleCI log files.

The build was using an OP2Utility version that was 6 commits ahead of the recorded SHA-1 for the submodule. Hence, it was not building with the intended version.
